x-pack/plugin/esql/qa/testFixtures/src/main/resources/tbucket.csv-spec

```diff
@@ -346,3 +346,24 @@ day_total:long | hour_med:double | h:datetime               | message:keyword
 1              | 1756467.0       | 2023-10-23T13:00:00.000Z | Connected to 10.1.0.1
 1              | 1232382.0       | 2023-10-23T13:00:00.000Z | Disconnected
 ;
+
+
+tbucketWithTimezone
+required_capability: date_trunc_timezone_support
+
+SET time_zone = "+05:00"\;
+FROM employees
+| RENAME hire_date as @timestamp
```

Comment:
I know this is tangential to this PR, but we need to be very careful with queries that rename timestamp in the implicit timestamp functions.  For `TBucket` in a `FROM` context, this is probably okay, but this would break `Rate`, since it depends on the index being sorted by the `@timestamp` field.  It might be good to generally forbid shadowing `@timestamp` on queries that use implicit timestamps.

I bring this up because this behavior is still in flux, and we might not want to build tests that depend on it behaving one specific way.